<launch>
  <!-- Interfaces -->
  <arg name="input_route" />
  <arg name="input_occupancy_grid" />
  <arg name="input_scenario" />
  <arg name="input_twist" default="/vehicle/status/twist" />
  <arg name="output_trajectory" />
  <arg name="is_completed" />

  <!-- Node configurations -->
  <arg name="waypoints_velocity" default="5.0" />
  <arg name="update_rate" default="10.0" />
  <arg name="th_arrived_distance_m" default="3.0" />
  <arg name="th_stopped_time_sec" default="1.0" />
  <arg name="th_stopped_velocity_mps" default="0.01" />
  <arg name="th_course_out_distance_m" default="5.0" />

  <!-- A* search configurations -->
  <arg name="use_back" default="true" />
  <arg name="use_potential_heuristic" default="false" />
  <arg name="use_wavefront_heuristic" default="false" />
  <arg name="time_limit" default="30000.0" />
  <arg name="robot_length" default="4.5" />
  <arg name="robot_width" default="1.75" />
  <arg name="robot_base2back" default="1.0" />
  <arg name="minimum_turning_radius" default="9.0" />
  <arg name="theta_size" default="144" />
  <arg name="curve_weight" default="1.2" />
  <arg name="reverse_weight" default="2.00" />
  <arg name="lateral_goal_range" default="0.5" />
  <arg name="longitudinal_goal_range" default="2.0" />
  <arg name="angle_goal_range" default="6.0" />
  <arg name="obstacle_threshold" default="100" />
  <arg name="potential_weight" default="10.0" />
  <arg name="distance_heuristic_weight" default="1.0" />

  <node pkg="freespace_planner" type="astar_navi" name="astar_navi" output="screen">
    <remap from="~input/route" to="$(arg input_route)" />
    <remap from="~input/occupancy_grid" to="$(arg input_occupancy_grid)" />
    <remap from="~input/scenario" to="$(arg input_scenario)" />
    <remap from="~input/twist" to="$(arg input_twist)" />
    <remap from="~output/trajectory" to="$(arg output_trajectory)" />
    <remap from="~is_completed" to="$(arg is_completed)" />

    <param name="waypoints_velocity" value="$(arg waypoints_velocity)" />
    <param name="update_rate" value="$(arg update_rate)" />
    <param name="th_arrived_distance_m" value="$(arg th_arrived_distance_m)" />
    <param name="th_stopped_time_sec" value="$(arg th_stopped_time_sec)" />
    <param name="th_stopped_velocity_mps" value="$(arg th_stopped_velocity_mps)" />

    <param name="use_back" value="$(arg use_back)" />
    <param name="use_potential_heuristic" value="$(arg use_potential_heuristic)" />
    <param name="use_wavefront_heuristic" value="$(arg use_wavefront_heuristic)" />
    <param name="time_limit" value="$(arg time_limit)" />
    <param name="robot_length" value="$(arg robot_length)" />
    <param name="robot_width" value="$(arg robot_width)" />
    <param name="robot_base2back" value="$(arg robot_base2back)" />
    <param name="minimum_turning_radius" value="$(arg minimum_turning_radius)" />
    <param name="theta_size" value="$(arg theta_size)" />
    <param name="angle_goal_range" value="$(arg angle_goal_range)" />
    <param name="curve_weight" value="$(arg curve_weight)" />
    <param name="reverse_weight" value="$(arg reverse_weight)" />
    <param name="lateral_goal_range" value="$(arg lateral_goal_range)" />
    <param name="longitudinal_goal_range" value="$(arg longitudinal_goal_range)" />
    <param name="obstacle_threshold" value="$(arg obstacle_threshold)" />
    <param name="potential_weight" value="$(arg potential_weight)" />
    <param name="distance_heuristic_weight" value="$(arg distance_heuristic_weight)" />
  </node>
</launch>
